### ISSUE TITLE
remove_invalid() makes object even more invalid

### ISSUE THREAD

**Issue by cifkao** [salu133445/muspy#49]
Check this out:
```python
>>> m = muspy.Music(tracks=[muspy.Track(notes=[
    muspy.Note(time=-2, pitch=58, duration=19, velocity=99),
    muspy.Note(time=-1, pitch=65, duration=20, velocity=65),
    muspy.Note(time=0, pitch=67, duration=23, velocity=65),
])])
>>> m.remove_invalid()
Music(metadata=Metadata(schema_version='0.0'), resolution=24, tracks=[Note(time=0, pitch=67, duration=23, velocity=65)])
```
The valid note somehow gets added to the track list, and the track itself gets removed.

Not sure why this is happening, but #47 seems to fix it.

**Comment by salu133445** [comment]
Looks like a bug. Will check the pull request. Thanks.

**Comment by salu133445** [comment]
This has been fixed by 8899576.

Note: When applying recursively, we have to modifying the lists in place so that their references remain unchanged.
